src/transform/tsTransform.ts

```diff
@@ -1,86 +1,31 @@
-  if (callExpression.kind !== ts.SyntaxKind.CallExpression) return false;
-  const { expression, arguments: args } = callExpression as ts.CallExpression;
-  if (expression.kind !== ts.SyntaxKind.Identifier) {
-    return false;
-  }
-  const text = (expression as ts.Identifier).escapedText as string;
-  if (!['require', devImports.variable].includes(text)) {
-    return false;
-  }
-  if (args.length !== 1) return false;
-  if (ts.isStringLiteralLike(args[0])) {
-    return true;
-  }
-}
-
-export interface ITypescriptTransformProps {
+export interface ITypescriptTransformProps extends ts.TranspileOptions {
   input: string;
-  fileName?: string;
-  compilerOptions?: ts.CompilerOptions;
-
   webWorkers?: Array<IWebWorkerItem>;
   replacements?: IStatementReplaceableCollection;
 }
 
-export function visitStatementNode(node, replacer: (input) => any) {
-  if (isRequireCall(node)) {
-    const replacement = replacer(node.arguments[0].text);
-    if (replacement) {
-      node.arguments[0] = ts.createStringLiteral(replacement);
-    }
-  }
-}
 export function tsTransform(props: ITypescriptTransformProps): ts.TranspileOutput {
-  function moduleTransformer<T extends ts.Node>(): ts.TransformerFactory<T> {
-    return context => {
-      const visit: ts.Visitor = node => {
-        if (props.webWorkers) {
-          if (ts.isNewExpression(node)) {
-            const newText = node.expression.getText();
-            if (newText === 'Worker' || newText === 'SharedWorker') {
-              if (node.arguments.length === 1) {
-                const firstArg = node.arguments[0];
-                const statement = firstArg['text'];
-                // map item
-                const item = props.webWorkers.find(item => {
-                  return item.path === statement && item.type === newText;
-                });
-                if (item) {
-                  return ts.createNew(ts.createIdentifier(newText), undefined, [
-                    ts.createStringLiteral(item.bundlePath),
-                  ]);
-                }
-              }
-            }
-          }
-        } else {
-          visitStatementNode(node, statement => {
-            const replacement = props.replacements.find(item => item.fromStatement === statement);
-            if (replacement) {
-              return replacement.toStatement;
-            }
-          });
-        }
-
-        return ts.visitEachChild(node, child => visit(child), context);
-      };
+  const after = [];
 
-      return node => ts.visitNode(node, visit);
-    };
+  if (props.replacements || props.webWorkers) {
+    after.push(moduleTransformer(props));
   }
 
-  const after = [];
-  if (props.replacements || props.webWorkers) {
-    after.push(moduleTransformer());
+  if (!props.transformers) {
+    props.transformers = {};
   }
+
   return ts.transpileModule(props.input, {
     fileName: props.fileName,
     compilerOptions: props.compilerOptions,
-    transformers: { after: after },
+    transformers: {
```

Comment:
Do you think it makes sense to create a function `tsTranspilerModule` and pass properties. This way we could also test it. There are 2 places where we call `ts.transpileModule`